src/oic/oic/__init__.py

```diff
@@ -1319,6 +1345,12 @@ def create_registration_request(self, **kwargs):
         except KeyError:
             pass
 
+        try:
```

Comment:
I think it is better to change `try .. except`  to a condition `if response_types in req`.

Reply:
I'm OK with either or so we'll go for your proposal.